### PR TITLE
[ADLS][Samples]Add Samples for sync and async

### PR DESCRIPTION
**Pull request by xiafu-msft**

### CODE REVIEW

**Comment by adxsdk6** [comment]
Can one of the admins verify this patch?

File: sdk/storage/azure-storage-blob/samples/blob_samples_containers.py [hunk 1]
```diff
@@ -44,7 +45,7 @@ def container_sample(self):
         # [START create_container_client_sasurl]
         from azure.storage.blob import ContainerClient
 
-        sas_url = sas_url = "https://account.blob.core.windows.net/mycontainer?sv=2015-04-05&st=2015-04-29T22%3A18%3A26Z&se=2015-04-30T02%3A23%3A26Z&sr=b&sp=rw&sip=168.1.5.60-168.1.5.70&spr=https&sig=Z%2FRHIX5Xcg0Mq2rqI3OlWTjEg2tYkboXr1P9ZUXDtkk%3D"
+        sas_url = "https://account.blob.core.windows.net/mycontainer?sv=2015-04-05&st=2015-04-29T22%3A18%3A26Z&se=2015-04-30T02%3A23%3A26Z&sr=b&sp=rw&sip=168.1.5.60-168.1.5.70&spr=https&sig=Z%2FRHIX5Xcg0Mq2rqI3OlWTjEg2tYkboXr1P9ZUXDtkk%3D"
```
Comment (rakshith91) [inline sdk/storage/azure-storage-blob/samples/blob_samples_containers.py]: Although, I'm certain these are fake values, do you think it;s a good idea for us to generate sas url / import from env variable in the samples?

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py [hunk 2]
```diff
@@ -176,6 +184,15 @@ def delete_directory(self, **kwargs):
         :keyword int timeout:
             The timeout parameter is expressed in seconds.
         :return: None
+
+        .. admonition:: Example:
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py]: I'm sure you did, but please make sure you generate docs locally and they seem fine.

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_generated/aio/_data_lake_storage_client_async.py [hunk 3]
```diff
@@ -64,3 +64,9 @@ def __init__(
         return self
     async def __aexit__(self, *exc_details):
         await self._client.__aexit__(*exc_details)
+
+    async def close(self):
+        """ This method is to close the sockets opened by the client.
+        It need not be used when using with a context manager.
+        """
+        await self._client.close()
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_generated/aio/_data_lake_storage_client_async.py]: nit: end of line

File: sdk/storage/azure-storage-file-datalake/samples/datalake_samples_access_control_async.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+import asyncio
+import os
+import random
+import uuid
+
+from azure.storage.filedatalake.aio import (
+    DataLakeServiceClient,
+)
+
```
Comment (rakshith91) [inline sdk/storage/azure-storage-file-datalake/samples/datalake_samples_access_control_async.py]: All sample files must have docstrings in this format; for example
https://github.com/Azure/azure-sdk-for-python/blob/master/sdk/storage/azure-storage-blob/samples/blob_samples_common.py#L10-L17

File: sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py (hunk 2 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_data_lake_directory_client.py]: Thanks for reminding me that! 
 Just noticed some of them look weird file:///C:/Users/xiafu/Downloads/azure-storage-file-datalake%20(1)/azure.storage.filedatalake.html

File: sdk/storage/azure-storage-blob/samples/blob_samples_containers.py (hunk 1 above)
Comment (xiafu-msft) [reply sdk/storage/azure-storage-blob/samples/blob_samples_containers.py]: I think the fake one is pretty straight forward. ^_^
Also when this code snippet is to tell user how to instantiate client using sas url directly

**Pull request merged**